src/cares.c

```diff
@@ -486,16 +490,37 @@ query_txt_cb(void *arg, int status,int timeouts, unsigned char *answer_buf, int
         goto callback;
     }
 
+    first_record = 1;
+    assembled_txt = strdup("");
```

Comment:
don't use functions that allocate the memory by themselves, we want to use the Python mamory allocator. You can create a string with https://docs.python.org/2/c-api/string.html#c.PyString_FromString and then resize it with https://docs.python.org/2/c-api/string.html#c._PyString_Resize